gulp builds give misleading errors on unsuccessful tasks

I tried running `gulp runtests-parallel`.

```
src/testRunner/parallel/host.ts:10:51 - error TS7016: Could not find a declaration file for module 'ms'. '/home/daniel/shared/TypeScript3/node_modules/ms/index.js' implicitly has an 'any' type.
  Try `npm install @types/ms` if it exists or add a new declaration (.d.ts) file containing `declare module 'ms';`

10         const ms = require("ms") as typeof import("ms");
                                                     ~~~~

yarn

Found 1 error.

[10:54:08] 'buildRules' errored after 1.1 min
[10:54:08] Error: Process exited with code: 1
    at ChildProcess.proc.on.exitCode (/home/daniel/shared/TypeScript3/scripts/build/utils.js:52:24)
    at ChildProcess.emit (events.js:182:13)
    at ChildProcess.EventEmitter.emit (domain.js:460:23)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:237:12)
[10:54:08] 'runtests-parallel' errored after 1.12 min
[10:54:08] The following tasks did not complete: buildTests, <series>, <series>, buildTypescriptServicesOut, buildServerLibraryOut
[10:54:08] Did you forget to signal async completion?
```

Is it possible that the build file forgot to signal async completion?

--- a/scripts/build/tests.js
+++ b/scripts/build/tests.js
@@ -140,13 +140,8 @@ async function runConsoleTests(runJs, defaultReporter, runInParallel, watchMode,
     await deleteTemporaryProjectOutput();
 
     if (error !== undefined) {
-        if (watchMode) {
-            throw error;
-        }
-        else {
-            log.error(error);
-            process.exit(typeof errorStatus === "number" ? errorStatus : 2);
-        }
+        process.exitCode = typeof errorStatus === "number" ? errorStatus : 2;
+        throw error;
     }
 }
 exports.runConsoleTests = runConsoleTests;

--- a/scripts/build/utils.js
+++ b/scripts/build/utils.js
@@ -340,6 +340,7 @@ function rm(dest, opts) {
             duplex.push(file);
             cb();
         }
+        duplex.push(null); // signal end of read queue
     };
 
     const duplex = new Duplex({

